packages/ui/components/badge/Badge.tsx

```diff
@@ -20,14 +21,24 @@ const classNameBySize = {
   lg: "h-6",
 };
 
-export type BadgeProps = {
+export type BadgeBaseProps = {
   variant: keyof typeof badgeClassNameByVariant;
   size?: keyof typeof classNameBySize;
   StartIcon?: SVGComponent;
   bold?: boolean;
   withDot?: boolean;
   rounded?: boolean;
-} & JSX.IntrinsicElements["div"];
+};
+
+export type BadgeProps =
```

Comment:
Updated badge component to also be a button instead of a clickable div. This button-badge is better for a11y.